File: back/src/main/java/com/accountingapi/service/QuotationService.java
package com.accountingapi.service;

import com.accountingapi.model.Purchase;
import com.accountingapi.model.Quotation;

import java.util.List;

public interface QuotationService {

    Quotation addQuotation(Quotation quotation);

    List<Quotation> findAllQuotations();

    Quotation findQuotationById(Long quotationId);

    void deleteQuotationById(Long quotationId);

    boolean existsById(Long quotationId);
}

File: front/src/app/_services/quotation.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BASE_URL, QUOTATIONS} from '../_globals/vars';

@Injectable({
  providedIn: 'root'
})
export class QuotationService {

  constructor(private http: HttpClient) {
  }


  postQuotation(quotation) {
    return this.http.post(BASE_URL + QUOTATIONS, quotation);
  }

  findAllQuotations() {
    return this.http.get(BASE_URL + QUOTATIONS);
  }


  findQuotationById(quotationId) {
    return this.http.get(BASE_URL + QUOTATIONS + quotationId);
  }

  deleteQuotationById(quotationId) {
    return this.http.delete(BASE_URL + QUOTATIONS + quotationId);
  }

  confirmQuotationById(quotationId) {
    return this.http.get(BASE_URL + QUOTATIONS + 'confirm/' + quotationId);
  }

  rejectQuotationById(quotationId) {
    return this.http.get(BASE_URL + QUOTATIONS + 'reject/' + quotationId);
  }

}
